tpu/utils/tpu_survival_job.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tpu_survival import TPUSurvival
+from time import sleep
+import sys
+
+project = 'PROJECT_ID'
+location = 'LOCATION'
+max_attempt = 10
+
+ts = TPUSurvival(project=project, location=location)
+
+
+while ts.current_index <= max_attempt:
+	# creatre TPU pod
+	ts.create()
+
+	# wait until it is ready
+	while ts.state != 'READY':
+		sleep(10)
+		ts.update_state()
+
+	ts.run_task()
```

Comment:
Does run_task generate an exception which need to be handled? otherwise top while loop may just terminate?

Reply:
I don't think so - `run_task` calls a subprocess to run the shell script which in turn runs the python training job; `run_task` itself returns immediate after the subprocess call.  That subprocess could fail for various reasons but those errors would not be propagated back to the `ts.run_task` call.

In the following while loop that subprocess is being polled and its returncode examined.

I can explore more robust ways of handling failing subprocesses, but at the same time do not want to make too much assumption on the user's environment.  Think of the `tpu_survival_job.py` as a basic workflow that has worked in the few experiments I ran, and hopefully have captured the majority of what could go wrong.